tests/ConfigTest.php

```diff
@@ -49,9 +66,17 @@ public static function configProvider(): array
         ];
     }
 
-    public function testItThrowsAnErrorIfAConfigFileWasInvalid()
+    public function testItThrowsAnErrorIfAConfigProviderReturnsNonArrayForMultiLevelKeys()
     {
-        $config = new Config(__DIR__ . '/config');
+        /** @var ConfigProvider&MockInterface */
```

Comment:
This typehint is missing the variable name. The proper hint would be `/** @var ConfigProvider&MockInterface $configProvider */`, but even that would be wrong in this context. It's also inferred anyway, so this can be removed.

Reply:
Variable name isn't required & VS code doesn't infer it correctly without it

Reply:
It's not strictly required but that's the standard. I also recommend changing to a non-shit IDE that can infer types properly.

Reply:
I'd recommend not getting caught up in what IDEs other people prefer